llpc/util/llpcElfWriter.cpp

```diff
@@ -823,6 +834,144 @@ Result ElfWriter<Elf>::getSectionTextShader(const char *name, const void **data,
   return Result::Success;
 }
 
+// =====================================================================================================================
+// Get the start pos of relocs used by Ps in the reloc section
+//
+// @param relocSection : Reloc section
+// @param psIsaOffset : Ps ISA offset in elf binary
+template <class Elf> size_t ElfWriter<Elf>::getRelocPsOffset(const SectionBuffer *relocSection, size_t psIsaOffset) {
+  size_t psStartPos = 0;
+
+  const size_t relocCount = static_cast<size_t>(relocSection->secHead.sh_size / relocSection->secHead.sh_entsize);
+
+  auto relocs = reinterpret_cast<typename Elf::Reloc *>(const_cast<uint8_t *>(relocSection->data));
+  while (psStartPos < relocCount) {
+    if (relocs[psStartPos].r_offset >= psIsaOffset)
+      break;
+    psStartPos++;
+  }
+
+  return psStartPos;
+}
+
+// =====================================================================================================================
+// Merge Reloc Section
+//
+// @param pFragmentRelocSection : Reloc section of fragement elf binary
+// @param fragmentIsaOffset : Ps ISA offset in fragement elf binary
+// @param pNonFragmentRelocSection : Reloc section of nonFragment elf binary
+// @param nonFragmentIsaOffset : Ps ISA offset in nonFragement elf binary
+// @param [out] pNewSection : Merged section
+template <class Elf>
+void ElfWriter<Elf>::mergeRelocSection(const SectionBuffer *pFragmentRelocSection, size_t fragmentIsaOffset,
+                                       const SectionBuffer *pNonFragmentRelocSection, size_t nonFragmentIsaOffset,
+                                       SectionBuffer *pNewSection) {
+  const unsigned fragmentRelocCount =
+      static_cast<unsigned>(pFragmentRelocSection->secHead.sh_size / pFragmentRelocSection->secHead.sh_entsize);
+
+  auto fragmentRelocs = reinterpret_cast<typename Elf::Reloc *>(const_cast<uint8_t *>(pFragmentRelocSection->data));
+
+  unsigned fragmentRelocPsStartPos = getRelocPsOffset(pFragmentRelocSection, fragmentIsaOffset);
+
+  // If fragment elf binary do not have relocs in Ps, then we do not need to merge
```

Comment:
nit: 'If the fragment elf binary does not have relocs in Ps`?

nit: 'then we do not need to merge' merge what?